From 701a501fb1c84296d2003c0092e4308742079591 Mon Sep 17 00:00:00 2001
From: David Kalnischkies <david@kalnischkies.de>
Date: Wed, 11 May 2022 18:02:07 +0200
Subject: [PATCH] Use temporary file instead of depending on /dev/stdin for
 triehash

/dev might not be populated for example in unshare chroots, so just
using a temporary file it is until triehash supports non-file input.

Regression-Of: f6438ea9e726a1c13ce8d90ac78cc272346ab0f8
Gbp-Dch: Ignore
---
 apt-pkg/CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/apt-pkg/CMakeLists.txt b/apt-pkg/CMakeLists.txt
index a40b041b8..469408e52 100644
--- a/apt-pkg/CMakeLists.txt
+++ b/apt-pkg/CMakeLists.txt
@@ -3,7 +3,8 @@ include_directories(${PROJECT_BINARY_DIR}/include/apt-pkg)
 
 file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/include/apt-pkg/)
 execute_process(COMMAND grep -v "^#" "${CMAKE_CURRENT_SOURCE_DIR}/tagfile-keys.list"
-                COMMAND ${TRIEHASH_EXECUTABLE}
+		OUTPUT_FILE "${CMAKE_CURRENT_BINARY_DIR}/tagfile-keys.list")
+execute_process(COMMAND ${TRIEHASH_EXECUTABLE}
                         --ignore-case
                          --header ${PROJECT_BINARY_DIR}/include/apt-pkg/tagfile-keys.h
                          --code ${CMAKE_CURRENT_BINARY_DIR}/tagfile-keys.cc
@@ -12,7 +13,7 @@ execute_process(COMMAND grep -v "^#" "${CMAKE_CURRENT_SOURCE_DIR}/tagfile-keys.l
                          --function-name pkgTagHash
                          --include "<apt-pkg/tagfile.h>"
                          --include "<apt-pkg/header-is-private.h>"
-                         /dev/stdin )
+                         "${CMAKE_CURRENT_BINARY_DIR}/tagfile-keys.list")
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS "tagfile-keys.list")
 set_source_files_properties("${CMAKE_CURRENT_BINARY_DIR}/tagfile-keys.cc" PROPERTIES COMPILE_DEFINITIONS APT_COMPILING_APT)
 
-- 
GitLab

